web/src/message_fetch.ts

```diff
@@ -715,6 +720,54 @@ export function initialize(finished_initial_fetch: () => void): void {
         } catch (error) {
             blueslip.error("Error in recent_view_ui.process_messages", undefined, error);
         }
+
+        if (
+            !recent_view_ui.is_backfill_in_progress &&
+            !first_messages_fetch &&
+            initial_backfill_for_all_messages_done
+        ) {
+            // We only populate other cached data for major backfills.
+            return;
+        }
```

Comment:
I'm not sure what motivations this check -- is it to avoid wasting resources, or something else?

Reply:
Yeah, to avoid backfilling and rerendering on every new message sent.